### ISSUE TITLE
Consider using abstraction layer for SQL query generation

### ISSUE THREAD

**Issue by epatters** [AlgebraicJulia/AlgebraicRelations.jl#7]
I tend to be leery about generating code by string manipulation. We should consider use a DSL/AST for generating SQL queries. I just found out about [Octo.jl](https://github.com/wookay/Octo.jl), which looks nice.

**Comment by bosonbaas** [comment]
Yeah, the string manip to generate SQL queries is pretty hacky. I'll look into getting some kind of DSL in there to generate queries, and Octo.jl looks really nice for that, thanks!

**Comment by epatters** [comment]
On the Julia Zulip, I found out about the new package [FunSQL.jl](https://github.com/MechanicalRabbit/FunSQL.jl), which looks like it provides a SQL AST and pretty-printer.

**Comment by jpfairbanks** [comment]
I've talked with the author of that package a few times and I think we are aligned in the goals of making higher level querying interfaces. So it makes sense to try and use that package.
